extern/CMakeProject-mad.cmake

```diff
@@ -1,52 +1,88 @@
-list(APPEND MAD_SRC
-  "mad-0.15.1b/bit.c"
-  "mad-0.15.1b/decoder.c"
-  "mad-0.15.1b/fixed.c"
-  "mad-0.15.1b/frame.c"
-  "mad-0.15.1b/huffman.c"
-  "mad-0.15.1b/layer12.c"
-  "mad-0.15.1b/layer3.c"
-  "mad-0.15.1b/stream.c"
-  "mad-0.15.1b/synth.c"
-  "mad-0.15.1b/timer.c"
-  "mad-0.15.1b/version.c"
+set(MAD_DIR "${SM_EXTERN_DIR}/libmad-0.15.1b")
+
+check_include_file(assert.h HAVE_ASSERT_H)
+check_include_file(dlfcn.h HAVE_DLFCN_H)
+check_include_file(errno.h HAVE_ERRNO_H)
+check_function_exists(fcntl HAVE_FCNTL)
+check_function_exists(fork HAVE_FORK)
+check_include_file(inttypes.h HAVE_INTTYPES_H)
+check_include_file(limits.h HAVE_LIMITS_H)
+check_include_file(memory.h HAVE_MEMORY_H)
+check_function_exists(pipe HAVE_PIPE)
+check_include_file(stdlib.h HAVE_STDLIB_H)
+check_include_file(strings.h HAVE_STRINGS_H)
+check_include_file(string.h HAVE_STRING_H)
+check_include_file(sys/stat.h HAVE_SYS_STAT_H)
+check_include_file(sys/types.h HAVE_SYS_TYPES_H)
+check_include_file(sys/wait.h HAVE_SYS_WAIT_H)
+check_function_exists(waitpid HAVE_WAITPID)
+check_type_size(int SIZEOF_INT)
+check_type_size(long SIZEOF_LONG)
+check_type_size("long long" SIZEOF_LONG_LONG)
+check_include_files("stdlib.h;stdarg.h;string.h;float.h" STDC_HEADERS)
+
+configure_file("${MAD_DIR}/config.h.in" "${MAD_DIR}/config.h")
```

Comment:
If you didn't touch `config.h.in`, but merely used it as shown here, that is fine.